### PR TITLE
test: switch nav bar tests to react testing library

### PR DESCRIPTION
**Pull request by lsprr**
Issue #1159

This PR switches the Nav Bar tests to React Testing Library.

Validation steps

- Run yarn test
- See that there are coverage percentages in the table
- Verify that the tests are converted properly

### CODE REVIEW

File: packages/react/src/components/NavBar/navBar.test.tsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+      </NavItem>
+    </NavBar>
+  );
+
+  fireEvent.click(screen.getByRole('button'));
+  fireEvent.focusIn(screen.getByRole('listitem'));
+  expect(screen.getByRole('listitem')).toBeInTheDocument();
+});
+
+test('returns no axe violations', async () => {
```
Comment (scurker) [inline packages/react/src/components/NavBar/navBar.test.tsx:155]: Only thing missing is running axe against both collapsed/not-collapsed states.

**Review by scurker — CHANGES_REQUESTED** [review]
Sorry I'm so far behind on these, but thank you again for your contribution! Just one small change needed here to test for an additional a11y state.

File: packages/react/src/components/NavBar/navBar.test.tsx (hunk 1 above)
Comment (lsprr) [reply packages/react/src/components/NavBar/navBar.test.tsx]: No worries! I understand you're very busy. Anything for you and Deque. 😄

**Review by scurker — APPROVED** [review]

**Comment by scurker** [comment]
🚨 Reviewed for Security 🚨

**Pull request merged**